app/javascript/packs/provider-dialogs-common.js

```diff
@@ -1,26 +1,127 @@
 import React from 'react';
-import renderModal from '../provider-dialogs/modal'
+import renderModal from '../provider-dialogs/modal';
 
-window.listenToRx(function(buttonData) {
-  if (!buttonData || buttonData.controller !== 'provider_dialogs') {
-    return;
-  }
+ManageIQ.angular.app.component('providerDialogUser', {
+  bindings: {
+    dialog: '<',
+    entityName: '@',
+    actionName: '@',
+  },
+  template: '<dialog-user dialog="$ctrl.dialog" refresh-field="$ctrl.refreshField(field)" on-update="$ctrl.setDialogData(data)"></dialog-user>',
+  controller: [function() {
+    let formData = {};
 
-  if (! buttonData.component_name) {
-    console.error('provider-dialogs: missing component_name', buttonData);
-    return;
-  }
+    /* this.refreshField = (field) => {
+      // dynamic fields are not supported in this context
+      console.log('TODO dialogFieldRefreshService.refreshField', field);
+      return Promise.resolve({values:[], options:[]});
+    }; */
+
+    this.setDialogData = (data) => {
+      formData = data.data;
+
+      ManageIQ.redux.store.dispatch({
+        type: 'FormButtons.saveable',
+        payload: data.validations.isValid,
+      });
+    };
 
-  reactModal(buttonData);
+    this.$onInit = () => {
+      ManageIQ.redux.store.dispatch({
+        type: 'FormButtons.init',
+        payload: {
+          newRecord: true,
+          pristine: false,
+          addClicked: () => {
+            apiCall(
+              {
+                entity_name: this.entityName,
```

Comment:
either this is missing `success_message`, or we need to have some default for that